src/mechanics/solver/EigenSolver.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include <string>
+#include "Eigen/Sparse"
+
+namespace NuTo
+{
+
+//! Solve a sparse linear system \f$A x = b\f$ made of Eigen types.
+//! @param A Sparse matrix.
+//! @param b Right hand side vector.
+//! @param solver String representation of solver.
+//!
+//! Choose among the following solvers:
+//!
+//! Built-in direct solvers:
+//! - `EigenSparseLU`
+//! - `EigenSparseQR`
+//! - `EigenSimplicialLLT`
+//! - `EigenSimplicialLDLT`
+//! 
+//! Built-in iteratitive solvers:
+//! - `EigenConjugateGradient`
+//! - `EigenLeastSquaresConjugateGradient`
+//! - `EigenBiCGSTAB.
+//! 
+//! and [SuiteSparse](http://faculty.cse.tamu.edu/davis/suitesparse.html) solvers (need to be installed seperately):
+//! - `EigenUmfPackLU`
+//! - `EigenCholmodSupernodalLLT`
+Eigen::VectorXd EigenSolver(Eigen::SparseMatrix<double> A, Eigen::VectorXd b, std::string solver);
```

Comment:
1. Kay.
2. Could do, or separate the SuiteSparse solvers out as well, so that this file doesn't become too big; what do you think?
3. Kay.
4. a) Enum is annoying in Python again.
    b) What happens if e.g. I don't have SuiteSparse installed? Throw or Enum value is not available?

Reply:
2) This wrapper allows easy swap of the solvers without changing the includes. Otherwise I just include them in my main file manually. So I vote for including all solvers that have the eigen sparse solver concept. 
3) [_thanks, markdown enumeration numbering._]
4) We try hard to not base or design on a possible python interface, right? :ghost: And even in python, I would be happy to have some sort of autocompletion instead of guessing the solver name (and its capitalization - well, that could be solved...)